libraries/chain/database.cpp

```diff
@@ -246,22 +246,46 @@ bool database::is_known_transaction( const transaction_id_type& id )const
    return trx_idx.find( id ) != trx_idx.end();
 } FC_CAPTURE_AND_RETHROW() }
 
-block_id_type database::get_block_id_for_num( uint32_t block_num )const
+block_id_type database::find_block_id_for_num( uint32_t block_num )const
 {
    try
    {
+      if( block_num == 0 )
+         return block_id_type();
+
+      // Reversible blocks are *usually* in the TAPOS buffer.  Since this
+      // is the fastest check, we do it first.
+      block_summary_id_type bsid = block_num & 0xFFFF;
```

Comment:
Fair enough, the code is correct and we can change it later. We should probably do an audit of the codebase for magic numbers as it is. I'll create a ticket for it.